qemu/tests/migration_with_json_backing.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+import json
+
+from avocado.utils import process
+
+from virttest.qemu_storage import QemuImg, get_image_json, get_image_repr
+from virttest.data_dir import get_data_dir
+from virttest.qemu_devices.qdevices import QBlockdevFormatNode
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+def run(test, params, env):
+    """
+    KVM migration test:
+    1) Create a snapshot chain base->sn1->sn2 with backing:json format.
+    2) Start src guest with sn2.
+    3) Start dst guest with sn2 and listening status.
+    4) Do migration from src to dst.
+    5) After migration finished, check block info in dst.
+    6) In dst, do snapshot from sn2 to sn3
+    7) Check sn3 backing chain by qemu-img info
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def get_img_objs(images):
+        return [QemuImg(params.object_params(tag), get_data_dir(), tag)
+                for tag in images]
+
+    def _create_image_with_backing(image, json_format=False):
```

Comment:
1. if image_format_image1 = luks, this function may not work correctly
2. for json_format=False, we can use VT function (image.create) to create image
3. if image_format_image1 = luks, we can also use image.create to create image, for in this case, image1 will be represented in json format